btp/provider/resource_subaccount_destination_certificate.go

```diff
@@ -0,0 +1,300 @@
+						MarkdownDescription: "The time unit associated with the validity duration, such as `DAYS`, `MONTHS`, or `YEARS`.",
+						Computed:            true,
+						PlanModifiers: []planmodifier.String{
+							stringplanmodifier.UseStateForUnknown(),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (rs *subaccountDestinationCertificateResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+
+	var data subaccountDestinationCertificateResourceType
+
+	diags := req.State.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cliRes, _, err := rs.cli.Connectivity.DestinationCertificate.Get(ctx, &btpcli.DestinationCertificateGetInput{
+		SubaccountId:      data.SubaccountId.ValueString(),
+		ServiceInstanceId: data.ServiceInstanceId.ValueString(),
+		CertificateName:   data.CertificateName.ValueString(),
+	})
+	if err != nil {
+		resp.Diagnostics.AddError("API Error Reading Resource Destination Certificate", fmt.Sprintf("%s", err))
+		return
+	}
+
+	state, diags := subaccountDestinationCertificateValueFrom(ctx, cliRes)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	state.SubaccountId = data.SubaccountId
+	state.ServiceInstanceId = data.ServiceInstanceId
+	state.CertificateContent = data.CertificateContent
+
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+}
+
+func (rs *subaccountDestinationCertificateResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	var plan subaccountDestinationCertificateResourceType
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	cliRes, _, err := rs.cli.Connectivity.DestinationCertificate.Create(ctx, &btpcli.DestinationCertificateCreateInput{
+		SubaccountId:      plan.SubaccountId.ValueString(),
+		ServiceInstanceId: plan.ServiceInstanceId.ValueString(),
+		Certificate: btpcli.FileInput{
+			Filename:           plan.CertificateName.ValueString(),
+			CertificateContent: plan.CertificateContent.ValueString(),
+		},
+	})
+
+	if err != nil {
+		resp.Diagnostics.AddError("API Error Creating Resource Destination Certificate", fmt.Sprintf("%s", err))
+		return
+	}
+
+	state, diags := subaccountDestinationCertificateValueFrom(ctx, cliRes)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	state.SubaccountId = plan.SubaccountId
+	state.ServiceInstanceId = plan.ServiceInstanceId
+	state.CertificateContent = plan.CertificateContent
+
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (rs *subaccountDestinationCertificateResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	resp.Diagnostics.AddError("Resource Destination Certificate does not support updates", "Destroy and recreate the resource")
```

Comment:
Can we make it easier for cusrtomers when enforcing replacement truggers in the schema?